src/supremm/datasource/pcp/pcpsummarize.py

```diff
@@ -0,0 +1,365 @@
+    def __init__(self, nodename, nodeidx, archivedata):
+        self._nodename = nodename
+        self._nodeidx = nodeidx
+        self._archivedata = archivedata
+
+    nodename = property(lambda self: self._nodename)
+    nodeindex = property(lambda self: self._nodeidx)
+    archive = property(lambda self: self._archivedata)
+
+class PCPSummarize(Summarize):
+    """
+    Summarize class is responsible for iteracting with the pmapi python code
+    and managing the calls to the various analytics to process the data
+    """
+
+    def __init__(self, preprocessors, analytics, job, config, fail_fast=False):
+        super().__init__(preprocessors, analytics, job, config, fail_fast)
+        self.start = time.time()
+        self.archives_processed = 0
+        self.rangechange = RangeChange(config)
+
+    def process(self):
+        """ Main entry point. All archives are processed """
+        success = 0
+        self.archives_processed = 0
+
+        for nodename, nodeidx, archive in self.job.nodearchives():
+            try:
+                self.processarchive(nodename, nodeidx, archive)
+                self.archives_processed += 1
+            except pmapi.pmErr as exc:
+                success -= 1
+                #pylint: disable=not-callable
+                self.adderror("archive", "{0}: pmapi.pmErr: {1}".format(archive, exc.message()))
+
+            except Exception as exc:
+                success -= 1
+                self.adderror("archive", "{0}: Exception: {1}. {2}".format(archive, str(exc), traceback.format_exc()))
+                if self.fail_fast:
+                    raise
+
+        return success == 0
+
+    def complete(self):
+        """ A job is complete if archives exist for all assigned nodes and they have
+            been processed sucessfullly
+        """
+        return self.job.nodecount == self.archives_processed
+
+    def good_enough(self):
+        """ A job is good_enough if archives for 95% of nodes have
+            been processed sucessfullly
+        """
+        return self.archives_processed >= 0.95 * float(self.job.nodecount)
+
+    def get(self):
+        """ Return a dict with the summary information """
+        output = {}
+        timeseries = {}
+
+        je = self.job.get_errors()
+        if len(je) > 0:
+            self.adderror("job", je)
+
+        if self.job.nodecount > 0:
+            for analytic in self.alltimestamps:
+                if analytic.status != "uninitialized":
+                    if analytic.mode == "all":
+                        output[analytic.name] = analytic.results()
+                    if analytic.mode == "timeseries":
+                        timeseries[analytic.name] = analytic.results()
+            for analytic in self.firstlast:
+                if analytic.status != "uninitialized":
+                    output[analytic.name] = analytic.results()
+
+        output['summarization'] = {
+            "version": VERSION,
```

Comment:
this could be why your ookami data is not showing.

Reply:
Resolved in:
https://github.com/ubccr/supremm/pull/259/commits/a34241f9050f4e6fcc5b9fb084199663ce3c9fe2
https://github.com/ubccr/supremm/pull/259/commits/e928967e850ddb22068e1e23ae33fa39e58e0b23
https://github.com/ubccr/supremm/pull/259/commits/c7b17840d6afddd9945104bdc48378a428543fb2